### PR TITLE
feat: add overrideable FavoriteToggle to core-components

### PR DESCRIPTION
**Pull request by callebstrom**
### Background
We need to allow customisation of the favorite icon color, while also standardising the pattern for marking something as "favorite" (currently used on entitiy rows, cards as well as scaffolder templates).

Closes https://github.com/backstage/backstage/issues/26385 

### Changes
- Add generic controlled component `FavoriteToggle`
- `FavoriteToggle` defaults to previous hard-coded yellow `#f3ba37`
- Expose icon separately as `FavoriteToggleIcon` for use with `MaterialTable`
- Replace all occurrences of favorite icon and tooltip with `FavoriteToggle`

### Screenshots
image
image
image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog/src/components/CatalogTable/CatalogTable.tsx [hunk 1]
```diff
@@ -160,12 +152,7 @@ export const CatalogTable = (props: CatalogTableProps) => {
 
       return {
         cellStyle: { paddingLeft: '1em' },
-        icon: () => (
-          <>
-            <Typography style={visuallyHidden}>{title}</Typography>
```
Comment (callebstrom) [inline plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:165]: Typography not required as tooltip achieves the same for SR:s

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Think that this is looking pretty good, but we just need to include a changeset for `core-components` :tada:

**Comment by callebstrom** [comment]
@benjdlambert strange, `yarn changset` did not include `core-components` for some reason. Added it now :+1:

**Review by benjdlambert — APPROVED** [review]
Nice! Just cleaned a few things up and skipped the prop type export for now to just inline it, less API surface.

**Comment by callebstrom** [comment]
@benjdlambert thank you!

**Pull request merged**